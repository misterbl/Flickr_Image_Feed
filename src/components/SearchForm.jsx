import React from "react";
import searchIcon from "../assets/search.svg";

class SearchForm extends React.PureComponent {
  onTagsChange = event => {
    const { setFieldValue } = this.props;
    setFieldValue("searchByTags", event.target.value);
  };

  render() {
    const { values, handleSubmit } = this.props;

    return (
      <form onSubmit={handleSubmit}>
        <button
          type="submit"
          id="next-button"
          className="absolute bn bg-transparent"
          disabled={values.searchByTags === ""}
        >
          <img src={searchIcon} alt="search icon" className="mt1" />
        </button>
        <label
          className="font-weight-bold d-flex justify-content-between"
          htmlFor="buildingUnit"
        />

        <input
          id="searchByTags"
          type="text"
          name="searchByTags"
          value={values.searchByTags}
          onChange={this.onTagsChange}
          placeholder="search by tags"
          className="pl4 bn h2"
        />
      </form>
    );
  }
}
export default SearchForm;
