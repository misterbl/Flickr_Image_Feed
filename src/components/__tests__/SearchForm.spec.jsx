import * as React from "react";
import { shallow } from "enzyme";
import SearchForm from "../SearchForm";

describe("SearchForm", () => {
  const props = {
    setFieldValue: jest.fn(),
    values: {
      searchByTags: ""
    }
  };
  const wrapper = shallow(<SearchForm {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls setFieldValue on input change", () => {
    wrapper.find("input").simulate("change", { target: { value: "value" } });
    expect(props.setFieldValue).toHaveBeenCalled();
  });
});
