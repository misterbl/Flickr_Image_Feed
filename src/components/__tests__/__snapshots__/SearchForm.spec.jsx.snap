// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    setFieldValue={[MockFunction]}
    values={
      Object {
        "searchByTags": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="absolute bn bg-transparent"
          disabled={true}
          id="search-by-tags-button"
          type="submit"
        >
          <img
            alt="search icon"
            className="mt1"
            src="search.svg"
          />
        </button>,
        <label
          className="font-weight-bold d-flex justify-content-between"
          htmlFor="searchByTags"
        />,
        <input
          className="pl4 br2 bn h2"
          id="searchByTags"
          name="searchByTags"
          onChange={[Function]}
          placeholder="search by tags"
          type="text"
          value=""
        />,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="search icon"
            className="mt1"
            src="search.svg"
          />,
          "className": "absolute bn bg-transparent",
          "disabled": true,
          "id": "search-by-tags-button",
          "type": "submit",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "search icon",
            "className": "mt1",
            "src": "search.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "font-weight-bold d-flex justify-content-between",
          "htmlFor": "searchByTags",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "pl4 br2 bn h2",
          "id": "searchByTags",
          "name": "searchByTags",
          "onChange": [Function],
          "placeholder": "search by tags",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="absolute bn bg-transparent"
            disabled={true}
            id="search-by-tags-button"
            type="submit"
          >
            <img
              alt="search icon"
              className="mt1"
              src="search.svg"
            />
          </button>,
          <label
            className="font-weight-bold d-flex justify-content-between"
            htmlFor="searchByTags"
          />,
          <input
            className="pl4 br2 bn h2"
            id="searchByTags"
            name="searchByTags"
            onChange={[Function]}
            placeholder="search by tags"
            type="text"
            value=""
          />,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="search icon"
              className="mt1"
              src="search.svg"
            />,
            "className": "absolute bn bg-transparent",
            "disabled": true,
            "id": "search-by-tags-button",
            "type": "submit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "search icon",
              "className": "mt1",
              "src": "search.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "font-weight-bold d-flex justify-content-between",
            "htmlFor": "searchByTags",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pl4 br2 bn h2",
            "id": "searchByTags",
            "name": "searchByTags",
            "onChange": [Function],
            "placeholder": "search by tags",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
